### PR TITLE
Update StreamingAOI notification to print change + edit counts per user

### PR DESCRIPTION
**Pull request by CloudNiner**
I had a miserable time emulating RDD.groupByKey in DataFrame. I'd be very curious if you have suggestions on how to avoid dropping to RDDs for this particular aggregation now that there's a working RDD implementation to compare against.

### CODE REVIEW

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala [hunk 1]
```diff
@@ -247,30 +259,39 @@ object StreamingAOIMonitor
...
 object AOIMonitorUtils extends Logging {
 
-  case class NotificationData(notificationId: String, userId: String, name: String, email: String)
+  case class NotificationUserChangeSummary(notificationId: String,
+                                           user: String,
+                                           editCount: Long,
+                                           changesetCount: Int)
 
   // notificationId, userId, geom, name, email
   type Notification = (String, String, jts.Geometry, String, String)
```
Comment (jpolchlo) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: Make this a case class.  Improves readability, allows better interaction with Dataset/DataFrame APIs.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala [hunk 2]
```diff
@@ -93,8 +93,10 @@ object StreamingAOIMonitor
 
             val notifications: List[Notification] = queryNotifications(interval)
             notifications.foreach { println(_) }
-            val notificationsDf = notifications
-              .toDF("notificationId", "userId", "geom", "name", "email")
+            val notificationsRdd = spark.sparkContext.parallelize(notifications).map {
```
Comment (jpolchlo) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: Don't pack into an RDD.  Use `spark.createDataset(notifications)` (requires that `notifications` are filled with a case class instance).  You can then `groupByKey(_.notificationId)` or `join` using a more DataFrame-like interface.

Comment (CloudNiner) [reply src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: Updated. This is definitely cleaner and clearer.

File: src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala [hunk 3]
```diff
@@ -174,35 +171,54 @@ object StreamingAOIMonitor
 
             // 2. EXTRACT SALIENT INFO FROM DIFFS
             //    Prepare a dataset of summaries, one for each notification to send.
-            val messageInfo: Dataset[NotificationSummary] = {
+            val changeSummary = {
               val stats = diffs.withDelta
                 .withColumn("measurements", DefaultMeasurements)
                 .withColumn("counts", DefaultCounts)
```
Comment (jpolchlo) [inline src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: Any thoughts about either using this or killing it?

Comment (CloudNiner) [reply src/analytics/src/main/scala/osmesa/analytics/oneoffs/StreamingAOIMonitor.scala]: Ah true. We're not using this at the moment. I suppose we just delete along with the commented out code and re-add later if that's what ends up being more useful.

**Pull request merged**